### PR TITLE
Implemented Monosingleton generic

### PR DESCRIPTION
**Pull request by gutaussehend-Harshal**

### CODE REVIEW

File: Assets/Scripts/EnemyTank.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
```
Comment (ahujarahul99) [inline Assets/Scripts/EnemyTank.cs:4]: Give NameSpace and give the class definition for the class.

File: Assets/Scripts/MonoSingletonGeneric.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
```
Comment (ahujarahul99) [inline Assets/Scripts/MonoSingletonGeneric.cs:4]: Give class defintion and give namespaces for the class

File: Assets/Scripts/PlayerTank.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
```
Comment (ahujarahul99) [inline Assets/Scripts/PlayerTank.cs:4]: Give class definition and give namespaces for the class

**Comment by gutaussehend-Harshal** [comment]
test

**Pull request merged**